esp-hal/src/timer/timg.rs

```diff
@@ -597,92 +354,41 @@ where
     }
 }
 
-#[doc(hidden)]
-pub trait Instance: Sealed {
-    fn register_block(&self) -> &RegisterBlock;
-
-    fn timer_group(&self) -> u8;
-
-    fn timer_number(&self) -> u8;
-
-    fn reset_counter(&self);
-
-    fn set_counter_active(&self, state: bool);
-
-    fn is_counter_active(&self) -> bool;
-
-    fn set_counter_decrementing(&self, decrementing: bool);
-
-    fn set_auto_reload(&self, auto_reload: bool);
-
-    fn set_alarm_active(&self, state: bool);
-
-    fn is_alarm_active(&self) -> bool;
-
-    fn load_alarm_value(&self, value: u64);
-
-    fn listen(&self);
-
-    fn unlisten(&self);
-
-    fn clear_interrupt(&self);
-
-    fn now(&self) -> u64;
-
-    fn divider(&self) -> u32;
-
-    fn set_divider(&self, divider: u16);
-
-    fn is_interrupt_set(&self) -> bool;
-}
-
 /// A timer within a Timer Group.
-pub struct TimerX<TG, const T: u8 = 0> {
-    phantom: PhantomData<TG>,
-}
+pub struct Timer {
+    /// Pointer to the register block for this TimerGroup instance.
+    pub register_block: *const RegisterBlock,
 
-impl<TG, const T: u8> Sealed for TimerX<TG, T> {}
+    /// The timer number inside the TimerGroup
+    pub timer: u8, // TODO should this be something else?
```

Comment:
I was wondering whether it should be `* const crate::peripherals::timg0::T` so that we don't have to keep calling `self.t()` within the functions.

I left this as a question for myself and then forgot :D, for some reason I used `// TODO` here instead of `FIXME` which is what I've been doing in the rest of this PR series.

Even though I think the compiler is smart enough to inline the `self.t()` calls, this probably should still be `* const crate::peripherals::timg0::T`, so I'll fix that now.

Reply:
We should probably do something with the PACs, referring to `timg0` in cases that may also be timg1 feels slightly off.

Reply:
When it came to changing this, it got a bit awkward. Plus storing two pointers is more memory than a pointer + two u8's, so I'm going to remove the TODO and leave it as is.